include/triton/Tools/LinearLayout.h

```diff
@@ -534,6 +534,15 @@ class LinearLayout {
     return reshapeOuts({{*getOutDimNames().begin(), getTotalOutDimSize()}});
   }
 
+  // Concatenates two layouts by their input dimensions. The layouts must have
+  // the same output dimensions and sizes and different input dimensions. The
+  // input dimensions of this layout are placed before those of 'other'.
+  [[nodiscard]] LinearLayout concatIns(const LinearLayout &other) const;
+  // Concatenates two layouts by their output dimensions. The layouts must have
+  // the same input dimensions and sizes and different output dimensions. The
+  // output dimensions of this layout are placed before those of 'other'.
+  [[nodiscard]] LinearLayout concatOuts(const LinearLayout &other) const;
```

Comment:
how are these different from `operator*`?

Reply:
concatIns does something like 

```
A.concatIns(B) ->
[ A ]
[ B ]

A.concatOuts(B) -> 
[ A | B ]

A * B -> 
[ A ? ]
[ ?  B ]
```

I'm not entirely sure what `operator*` does, tbh, but it is different than these two.

Reply:
Nit. I believe in your example the matrices are flipped (`A.concatIns(B) -> [A | B]`).

Right, writing proper documentation for `operator*`  is in my TODO list. Note that `A * B` does what you described if `A` and `B`  have disjoint ins and outs. If `A` and `B` have the same outs and disjoint ins, `A * B = [A | B]`. Similarly if they have disjoint outs but same ins.

I might be missing something here, but could you check whether `operator*` works for your use case? If not, perhaps providing a specific example in which they differ and what's the result of each of them would help solving this mistery

Reply:
Perhaps I didn't explain it properly in the documentation of the functions, but supposed I have

```
// Identity mapping from register->register
 - register=1 -> (1)
   register=2 -> (2)
   register=4 -> (4)
   register=8 -> (8)
   register=16 -> (16)
where out dims are: [register (size 32)]

// Zeros mapping from register->lane
 - register=1 -> (0)
   register=2 -> (0)
   register=4 -> (0)
   register=8 -> (0)
   register=16 -> (0)
where out dims are: [lane (size 32)]
```

I want to construct 

```
 - register=1 -> (1, 0)
   register=2 -> (2, 0)
   register=4 -> (4, 0)
   register=8 -> (8, 0)
   register=16 -> (16, 0)
where out dims are: [lane (size 32)]
```

But `operator*` gives

```
 - register=1 -> (1, 0)
   register=2 -> (2, 0)
   register=4 -> (4, 0)
   register=8 -> (8, 0)
   register=16 -> (16, 0)
   register=32 -> (0, 0)
   register=64 -> (0, 0)
   register=128 -> (0, 0)
   register=256 -> (0, 0)
   register=512 -> (0, 0)
where out dims are: [register (size 32), lane (size 32)]
```

Reply:
I guess I could have used `LinearLayout::identity1D(32, kRegister, kRegister) * LinearLayout::zeros1D(1, kRegister, kLane, 32)` but sometimes I'm trying to build a layout from less trivial quadrants. I.e. imagine I want to splice two layouts together `A(reg, lane) -> (reg, lane)` and `B(reg, lane) -> (reg, lane)` such that the resulting layout is

```
[ A.sublayout(reg, reg)   A.sublayout(reg, lane) ]
[ B.sublayout(lane, reg)  B.sublayout(lane, lane) ]
```